server/game/cards/08-MotC/AsahinaTakamori.js

```diff
@@ -0,0 +1,31 @@
+const DrawCard = require('../../drawcard.js');
+const { CardTypes, Durations, Players } = require('../../Constants');
+const AbilityDsl = require('../../abilitydsl');
+
+class AsahinaTakamori extends DrawCard {
+    setupCardAbilities() {
+        this.reaction({
+            title: 'Choose a character to pacify',
+            when: {
+                onCardPlayed: (event) => event.card.type === CardTypes.Character && event.card.isFaction('crane')
```

Comment:
You need to check that `context.player` is playing the card

Reply:
Good catch.  I've added a test and fix for that.